File: sdk/chocobase-go/storage.go
package chocobase

import "fmt"

// StorageClient manages object storage buckets and files.
type StorageClient struct {
	BaseURL string
	APIKey  string
}

func newStorageClient(baseURL, apiKey string) *StorageClient {
	return &StorageClient{
		BaseURL: baseURL + "/v1/storage/v1",
		APIKey:  apiKey,
	}
}

func (s *StorageClient) From(bucket string) *BucketClient {
	return &BucketClient{
		BaseURL: fmt.Sprintf("%s/object/%s", s.BaseURL, bucket),
		Bucket:  bucket,
		APIKey:  s.APIKey,
	}
}

type BucketClient struct {
	BaseURL string
	Bucket  string
	APIKey  string
}

type SignedURLResponse struct {
	SignedURL string `json:"signed_url"`
}

func (b *BucketClient) CreateSignedURL(path string, expiresIn int) (*SignedURLResponse, error) {
	return &SignedURLResponse{
		SignedURL: fmt.Sprintf("%s/sign/%s?expires_in=%d", b.BaseURL, path, expiresIn),
	}, nil
}

File: sdk/chocobase-rs/src/storage.rs
#[derive(Debug, Clone)]
pub struct BucketClient {
    pub base_url: String,
    pub bucket: String,
    pub api_key: String,
}

impl BucketClient {
    pub fn create_signed_url(&self, path: &str, expires_in: u64) -> String {
        format!("{}/sign/{}?expires_in={expires_in}", self.base_url, path)
    }
}

#[derive(Debug, Clone)]
pub struct StorageClient {
    pub base_url: String,
    pub api_key: String,
}

impl StorageClient {
    pub fn new(base_url: String, api_key: String) -> Self {
        Self {
            base_url: format!("{base_url}/v1/storage/v1"),
            api_key,
        }
    }

    pub fn from(&self, bucket: &str) -> BucketClient {
        BucketClient {
            base_url: format!("{}/object/{bucket}", self.base_url),
            bucket: bucket.to_string(),
            api_key: self.api_key.clone(),
        }
    }
}
